Thank you very much for the fix. I still get an error though. Here's the trace for you:

```
2024/12/10 07:59:48 decoded: UnregisterRemoteSKI([0xc0005d4030])
SENDING: {"jsonrpc":"2.0","id":42,"result":[]}
SENDING: {"jsonrpc":"2.0","method":"remote/RemoteSKIDisconnected","params":{"ski":"c259baf577b491f3b526d55eff930d3f3e6ae078"}}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xa9eaf1]

goroutine 726 [running]:
main.(*Remote).PropagateEvent(0xc0000d2410, {0xc0004d3cf0?, 0x126ff40?}, {0xc000152394, 0x28}, {0xd5ce58, 0xc0002085a0}, {0x0, 0x0}, {0xc5326a, ...})
        /home/-/Dokumente/ConsolinnoEnergy/eebus-go/examples/remote/ucs.go:49 +0x131
main.(*Remote).RegisterUseCase.func1({0xc000152394?, 0xd5ce58?}, {0xd5ce58?, 0xc0002085a0?}, {0x0?, 0x0?}, {0xc5326a?, 0xc00050ac80?})
        /home/-/Dokumente/ConsolinnoEnergy/eebus-go/examples/remote/ucs.go:32 +0x6d
github.com/enbility/eebus-go/usecases/usecase.(*UseCaseBase).removeDeviceFromAvailableEntityScenarios(0xc0001cc9c0, {0xd5ce58, 0xc0002085a0})
        /home/-/Dokumente/ConsolinnoEnergy/eebus-go/usecases/usecase/usecase.go:231 +0xa4
github.com/enbility/eebus-go/usecases/usecase.(*UseCaseBase).deviceOrEntityRemoved(0xc0003b0ff0?, {{0xc000152394, 0x28}, 0x0, 0x2, {0xd5ce58, 0xc0002085a0}, {0x0, 0x0}, {0x0, ...}, ...})
        /home/-/Dokumente/ConsolinnoEnergy/eebus-go/usecases/usecase/events.go:32 +0x55
github.com/enbility/eebus-go/usecases/usecase.(*UseCaseBase).HandleEvent(0xc0001cc9c0, {{0xc000152394, 0x28}, 0x0, 0x2, {0xd5ce58, 0xc0002085a0}, {0x0, 0x0}, {0x0, ...}, ...})
        /home/-/Dokumente/ConsolinnoEnergy/eebus-go/usecases/usecase/events.go:13 +0x58
created by github.com/enbility/spine-go/spine.(*events).Publish in goroutine 69
        /home/-/go/pkg/mod/github.com/enbility/spine-go@v0.0.0-20241209160856-1aed917e83e7/spine/events.go:100 +0x2d8
```

--- a/examples/remote/ucs.go
+++ b/examples/remote/ucs.go
@@ -45,8 +45,12 @@ func (r *Remote) PropagateEvent(
 ) {
 	params := make(map[string]interface{}, 2)
 	params["ski"] = ski
-	params["device"] = device.Address()
-	params["entity"] = entity.Address()
+	if device != nil {
+		params["device"] = device.Address()
+	}
+	if entity != nil {
+		params["entity"] = entity.Address()
+	}
 	for _, conn := range r.connections {
 		_ = conn.Notify(context.Background(), string(event), params)
 	}

--- a/usecases/usecase/usecase.go
+++ b/usecases/usecase/usecase.go
@@ -223,7 +223,7 @@ func (u *UseCaseBase) updateRemoteEntityScenarios(
 func (u *UseCaseBase) removeDeviceFromAvailableEntityScenarios(device spineapi.DeviceRemoteInterface) {
 	indicies := u.entityScenarioIndicesOfDevice(device)
 
-	for i := range indicies {
+	for _, i := range indicies {
 		u.removeEntityIndexFromAvailableEntityScenarios(i)
 	}
 

--- a/usecases/usecase/usecase_test.go
+++ b/usecases/usecase/usecase_test.go
@@ -43,6 +43,33 @@ func (s *UseCaseSuite) Test() {
 	assert.False(s.T(), result)
 }
 
+func (s *UseCaseSuite) Test_RemoveDeviceScenarios() {
+	result := s.uc.RemoteEntitiesScenarios()
+	assert.Equal(s.T(), 0, len(result))
+
+	scenarios := s.uc.AvailableScenariosForEntity(s.monitoredEntity)
+	assert.Equal(s.T(), 0, len(scenarios))
+
+	ok := s.uc.IsScenarioAvailableAtEntity(s.monitoredEntity, 1)
+	assert.False(s.T(), ok)
+
+	s.uc.updateRemoteEntityScenarios(s.monitoredEntity, []model.UseCaseScenarioSupportType{1, 2, 3})
+
+	result = s.uc.RemoteEntitiesScenarios()
+	assert.Equal(s.T(), 1, len(result))
+
+	scenarios = s.uc.AvailableScenariosForEntity(s.monitoredEntity)
+	assert.Equal(s.T(), 3, len(scenarios))
+
+	s.uc.removeDeviceFromAvailableEntityScenarios(s.monitoredEntity.Device())
+
+	result = s.uc.RemoteEntitiesScenarios()
+	assert.Equal(s.T(), 0, len(result))
+
+	scenarios = s.uc.AvailableScenariosForEntity(s.monitoredEntity)
+	assert.Equal(s.T(), 0, len(scenarios))
+}
+
 func (s *UseCaseSuite) Test_AvailableScenarios() {
 	result := s.uc.RemoteEntitiesScenarios()
 	assert.Equal(s.T(), 0, len(result))

